air/src/stack/stack_manipulation/mod.rs

```diff
@@ -0,0 +1,108 @@
+    // Given it is a degree 7 operation, 7 is added to all the individual constraints
+    // degree.
+
+    // There are two unique constraints in SWAP stack manipulation operation in which the first element
+    // in the current frame is swapped over with the second element.
+    8, 8,
+];
+
+// SYSTEM OPERATIONS TRANSITION CONSTRAINTS
+// ================================================================================================
+
+/// Builds the transition constraint degrees for the system operations.
+pub fn get_transition_constraint_degrees() -> Vec<TransitionConstraintDegree> {
+    CONSTRAINT_DEGREES
+        .iter()
+        .map(|&degree| TransitionConstraintDegree::new(degree))
+        .collect()
+}
+
+/// Returns the number of transition constraints for the system operations.
+pub fn get_transition_constraint_count() -> usize {
+    NUM_CONSTRAINTS
+}
+
+/// Enforces constraints for the stack manipulation operations.
+pub fn enforce_constraints<E: FieldElement>(
+    frame: &EvaluationFrame<E>,
+    result: &mut [E],
+    op_flag: &OpFlags<E>,
+) -> usize {
+    let mut index = 0;
+
+    let swap_op_flag = op_flag.swap();
+
+    // Enforce unique constaints of the SWAP operation.
+    index += enforce_swap_unique_constraints(frame, result, swap_op_flag);
+
+    index
+}
+
+// TRANSITION CONSTRAINT HELPERS
+// ================================================================================================
+
+/// Enforces unique constraints of the SWAP operation. The SWAP operation swaps the first
+/// two elements in the stack. Therefore, the following constraints are enforced:
+/// - The first element in the current frame should be equal to the second element in the
+///   next frame.
+/// - The second element in the current frame should be equal to the first element in the
+///   next frame.
+pub fn enforce_swap_unique_constraints<E: FieldElement>(
+    frame: &EvaluationFrame<E>,
+    result: &mut [E],
+    op_flag: E,
+) -> usize {
+    let mut constraint_offset = 0;
+
+    // Enforces the first element in the current frame is same as to the second element in the
+    // next frame
+    result[constraint_offset] = op_flag * are_equal(frame.selector(0), frame.selector_next(1));
+
+    constraint_offset += 1;
+
+    // Enforces the second element in the current frame is same as to the first element in the
+    // next frame
+    result[constraint_offset] = op_flag * are_equal(frame.selector(1), frame.selector_next(0));
+
+    constraint_offset += 1;
+
+    constraint_offset
+}
+
+trait EvaluationFrameExt<E: FieldElement> {
+    // --- Column accessors -----------------------------------------------------------------------
+
+    /// Gets the current element of the specified selector register in the stack.
+    fn selector(&self, index: usize) -> E;
+    /// Gets the next element of the specified selector register in the stack.
+    fn selector_next(&self, index: usize) -> E;
+}
```

Comment:
This seems to be exactly the same as in the stack constraints file. We should probably declare and implement this only once (probably in `air/src/stack/mod.rs` file).

Reply:
Moved it to `air/src/stack/mod.rs` and importing it in other logical operation units.